### DISCUSSION TITLE
"Ternary operator" in the json

### DISCUSSION THREAD

**Discussion by tawmoto** [nlohmann/json#3075] [Q&A] [answered]
Hi,

I want to know if I there is somehow of ternary operator in the json library, I know it sounds a bit silly, but maybe there is.

I have a `std::string str`

So a oneliner that replaces this code
```
if (!str.empty)
{
    json["name"] = str;
}
```

possible synthax:
```
{
  "pi": 3.141,
  "happy": true,
  "name": ?(!str.empty)
}
```
so if `str is empty` I expect
```
{
  "pi": 3.141,
  "happy": true,
}
```
```
if `str is non empty` I expect
{
  "pi": 3.141,
  "happy": true,
  "name": str
}
```

thank you

**Comment by nlohmann** [comment, marked as answer]
Unfortunately, it is not possible to overload the ternary operator in C++.
